plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSImportValidator.xtend

```diff
@@ -105,7 +107,25 @@ class N4JSImportValidator extends AbstractN4JSDeclarativeValidator {
 			}
 		}
 	}
-	
+
+	/**
+	 * We need jsx resources to depend on jsx backend. We are patching imports in the transpiler (to add the import to jsx backend if it is missing),
+	 * but transpiler will crash if that import will be invalid, i.e. project has no dependency on jsx backend. It would be ideal to add validation 
+	 * on manifest and not transpile, at least jsx files. Unfortunately changes to the manifest are a bit disconnected to changes of the individual 
+	 * files, e.g. adding jsx file does not trigger manifest validation. Also errors in the manifest do not prevent single file compilation. 
+	 * @see https://github.com/eclipse/n4js/issues/346
+	 */
+	@Check
+	def checkProjectDependsOnReact(Script script) {
```

Comment:
consider moving this method to class `N4JSXReactBindingValidator`.
Also: that class seems to contain all N4JSX-related validations, atm. So maybe we should rename it to just `N4JSXValidator`?